### PR TITLE
[IMP]survey: make certification reports editable by studio

### PR DESCRIPTION
**Pull request by angv-odoo**
The goal of this commit is to make certification reports editable by studio. We also move certification reports into separate report folder from views, and merge modern and classic views into one to avoid duplication.

Task-3148819

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/survey/report/survey_templates.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <template id="certification_report_view_general">
+            <!-- Style classes to be applyed to '#o_survey_certification': [no class](purple), gold, blue -->
+            <div id="o_survey_certification" t-att-data-oe-model="user_input._name" t-att-data-oe-id="user_input.id" t-att-class="'article certification-wrapper %s %s' % (layout_color,layout_template)">
+                <div class="certification">
+                    <div class="certification-seal" t-if="user_input.scoring_success and layout_template == 'modern'"/>
```
Comment (flch-odoo) [inline addons/survey/report/survey_templates.xml:8]: Maybe it was already discussed, but wouldn't it also reduce the complexity (in addition to helping extensibility) if we indeed introduced the `general` template but kept `_modern` and `_classic` as extensions?

File: addons/survey/report/survey_templates.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    </div>
+
+                    <div class="page certification-content">
+                        <div class="oe_structure"/>
+                        <div t-if="user_input.scoring_success">
+                         <div class="oe_structure"/>
+                            <p> <span>This certificate is presented to</span>
+                                <br/>
+                                <t t-set="certif_style" t-value="''"/>
+                                <t t-set="certified_name" t-value="user_input.partner_id.name or user_input.email"/>
```
Comment (flch-odoo) [inline addons/survey/report/survey_templates.xml:22]: Maybe you'd like to know and think about the discussion here: https://github.com/odoo/odoo/pull/113459.

**Review by flch-odoo — COMMENTED** [review]
Hello @angv-odoo ! 
Nice PR already! I have a couple suggestions/questions. As always, feel free to reach out :-)

File: addons/survey/report/survey_templates.xml (hunk 1 above)
Comment (angv-odoo) [reply addons/survey/report/survey_templates.xml:8]: Hey, thanks for checking the pr! Do you mean having _modern and _classic views still in the report separately from general template? 
If that's the case, I think it would add even more complexity no?  If user wants to make general changes in the template, they will have to change all three templates separately, and same goes for us, we would have to change code in 3 different places every time something changes. 
I think it should be better to have 2-3 t-if statements in the layout and save ourselves all the trouble (and trim a code a bit), but if something we can discuss it together with Frgi :)

**Comment by flch-odoo** [comment]
Yes, but `_modern` extends _general. It would only contain it's own logic
but use _general for the rest.
Are you saying that If we customize the general template in this case, the
extensions from _modern and _classic would not show them?

**Review by flch-odoo — COMMENTED** [review]
Hey @angv-odoo ! Questions and suggestions :)

File: addons/survey/static/src/scss/survey_reports.scss [hunk 3]
```diff
@@ -235,13 +235,14 @@
...
-                    transform: translateX(-50%);
-                }
+            }
+            
+            .certification-number {
+                width: 40mm;
+                bottom: 5mm;
+                left: 50%;
+                -webkit-transform: translateX(-50%);
+                transform: translateX(-50%);
```
[the remark this answers is not in the document]
Comment (angv-odoo) [reply addons/survey/static/src/scss/survey_reports.scss:245]: yes, without this piece of code certification-number in modern template "jumps out" from it's normal place

File: addons/survey/report/survey_templates.xml (hunk 1 above)
Comment (angv-odoo) [reply addons/survey/report/survey_templates.xml:8]: We figured out that there would be no point in having both templates in the end if I remember correctly.

File: addons/survey/static/src/scss/survey_reports.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (flch-odoo) [reply addons/survey/static/src/scss/survey_reports.scss:245]: As discussed, this is not a fix but required with the changes of this PR.

**Review by flch-odoo — COMMENTED** [review]
Hey @angv-odoo ,

One thing left though, please update the commit message to give a few more details about why merging the templates makes sense beyond reducing code duplication; i.e., that it is anyways working as a single template and instructions to only show specific content if the layout is modern will be required even on a distinct modern template (if I understood and summarized this correctly?)

Sidenote about the translations maybe left for the reviewer: I don't think it's necessary nor useful to apply this change as it will be done automatically as always in master, and the keys won't likely be matched to avoid re-translation in the target languages, but I may be wrong.

**Comment by flch-odoo** [comment]
Also, you need an upgrade script, and you can lowercase the `S` in  `[IMP] Survey`

File: addons/survey/views/survey_report_templates.xml [hunk 4]
```diff
@@ -1,122 +0,0 @@
...
-
-                    <div class="certification-content">
-                        <div t-if="user_input.scoring_success">
-                            <p>This certificate is presented to
-                                 <t t-set="certif_style" t-value="''"/>
-                                <t t-set="certified_name" t-value="user_input.partner_id.name or user_input.email or ''"/>
-                                <t t-if="certified_name.isupper()">
-                                    <t t-set="certif_style" t-value="certif_style + 'font-family: certification-serif;'"/>
-                                </t>
-                                <t t-if="len(certified_name) > 35">
```
Comment (awa-odoo) [inline addons/survey/views/survey_report_templates.xml:75]: Note that this was not present in the "modern" certification before your PR and now it is.
Not sure if a problem or not, would have to be tested (maybe you already did).

File: addons/survey/views/survey_report_templates.xml [hunk 5]
```diff
@@ -1,122 +0,0 @@
...
-                                    <t t-set="certif_style" t-value="certif_style + 'font-size: 20px; line-height: 4; font-family: certification-serif; '"/>
-                                </t>
-                                <t t-elif="len(certified_name) > 20">
```
Comment (awa-odoo) [inline addons/survey/views/survey_report_templates.xml:78]: I've been playing the "7 differences game" but it's hard to see what are the differences between the 2 layouts, that would justify that we had 2 before (and only 1 in your PR).

Did you notice any?

**Review by awa-odoo — COMMENTED** [review]
Hello @angv-odoo 

As discussed we can go with this solution then.
Alternatives do not sound better and it would be nice to allow people to customize this report.

It also works relatively well to modify one and have it apply for the others, it's only an issue when messing with colors / images (from what I have seen).
But extra text and customizing labels is nice already.

I left a few comments and questions before we can ship it.

Thanks & cheers!

File: addons/survey/views/survey_report_templates.xml (hunk 4 above)
Comment (angv-odoo) [reply addons/survey/views/survey_report_templates.xml:75]: not really, t-if="len(certified_name) > 20 and layout_template == 'modern' this part makes sure that template will never get to that part as long as it's "modern" certification

File: addons/survey/views/survey_report_templates.xml (hunk 5 above)
Comment (angv-odoo) [reply addons/survey/views/survey_report_templates.xml:78]: There were 3-4 small changes(placement of certification number, change in style for length of certified name and the badge), which I changed by either adding "layout_template" conditions or by making small changes into css. For now I believe that this template looks identical to the ones that are currently on runbot

File: addons/survey/report/survey_templates.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                            <p> <span>This certificate is presented to</span>
+                                <br/>
+                                <t t-set="certif_style" t-value="''"/>
+                                <t t-set="certified_name" t-value="user_input.partner_id.name or user_input.email or ''"/>
+                                <t t-if="certified_name.isupper()">
+                                    <t t-set="certif_style" t-value="certif_style + 'font-family: certification-serif;'"/>
+                                </t>
+                                <t t-if="len(certified_name) > 20 and layout_template == 'modern'">
+                                    <t t-set="certif_style" t-value="certif_style + 'font-size: 40px; line-height: 4;'"/>
+                                </t>
```
Comment (awa-odoo) [inline addons/survey/report/survey_templates.xml:29]: You're right, I did not read this part properly.

Maybe write it this way to make it a bit more readable?

```
if > 35 and classic:
    ...
elif > 20:
    if modern
        ....
    else:
        ....
```

?

Currently it's a bit hard to follow.

File: addons/survey/report/survey_templates.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <template id="certification_report_view_general">
+            <!-- Style classes to be applyed to '#o_survey_certification': [no class](purple), gold, blue -->
```
Comment (awa-odoo) [inline addons/survey/report/survey_templates.xml:5]: While you're at it, there is a small typo "applyed" :eyes:

Comment (angv-odoo) [reply addons/survey/report/survey_templates.xml]: @awa-odoo everything looks good now :+1:

**Comment by awa-odoo** [comment]
@robodoo r+

**Comment by awa-odoo** [comment]
@robodoo r+
